components/x-teaser/src/concerns/rules.js

```diff
@@ -5,15 +5,15 @@
 const rulesets = {
 	media: (props) => {
 		// If this condition evaluates to true then no headshot nor image will be displayed.
-		if (props.showVideo && props.video && props.video.url) {
+		if (props.showVideo && props.video?.url) {
 			return 'video'
 		}
 
-		if (props.showHeadshot && props.headshot && props.indicators.isColumn) {
+		if (props.showHeadshot && props.headshot && props.teaserMetadata?.indicators?.isColumn) {
 			return 'headshot'
 		}
 
-		if (props.showImage && props.image && props.image.url) {
+		if (props.showImage && props.mainImage?.url) {
```

Comment:
> How can this change support the legacy teaser type? Do we need to change the property name from image to mainImage after fetching content data from esClient in the consumer apps?

This will be a breaking change so yeah clients using the ElasticSearch API will have to update the field name if they want to migrate to the new major version of x-teaser. I have a [branch](https://github.com/Financial-Times/next-article/tree/rationalise-teaser) to do this in next-article as we're in the middle of migrating it to content-pipeline whilst still supporting content-es and the package page uses the x-teaser component directly rather than via [cp-content-pipeline-ui](https://github.com/Financial-Times/cp-content-pipeline/tree/main/packages/ui). However, there's no rush for other clients using ElasticSearch to update to this new major version of x-teaser, so this change needn't affect them.

> I also have a concern about the naming mainImage. It exists in esClient response and it's not equal to the teaser image.

Ah I didn't realise that the teaser image could be different, great point! I'm going to investigate now but we might have to represent this teaser image as `teaserMetadata.image` instead of `mainImage` then...

Reply:
After the breaking change is released, if we found bugs or x-dash components related to x-teaser need some changes, what are we going to do? Releasing two versions for v12 and v11? I wonder how we are going to maintain x-teaser's components (x-teaser, x-teaser-list and x-teaser-timeline).

Reply:
There [hasn't been much activity](https://github.com/Financial-Times/x-dash/commits/280b7c8201e4339cf19f931d6e2c50fb3c2efcb5/components/x-teaser) in the x-teaser package in the last few years so I don't expect there to be any bug fixes that will require changes in both versions, but you're right that it's a concern and we'll (the platforms team) be happy to backport any bug fixes if the need does arise.